R/sidebar.R

```diff
@@ -201,23 +202,18 @@ layout_sidebar <- function(
   max_height_mobile <- sidebar$max_height_mobile %||%
     if (is.null(height)) "250px" else "50%"
 
-  if (identical(sidebar$open, "always")) {
-    # sidebar initialization is only needed if collapsible
-    sidebar_id <- NULL
-    sidebar_init_js <- NULL
-  } else {
-    sidebar_id <- tagGetAttribute(sidebar$tag, "id")
-    sidebar_init_js <- sidebar_js_init(sidebar_id)
-  }
+  sidebar_init <- if (!identical(sidebar$open, "always")) TRUE
+  sidebar_border <- if (!is.null(border)) tolower(border)
+  sidebar_border_radius <- if (!is.null(border_radius)) tolower(border_radius)
 
   res <- div(
     class = "bslib-sidebar-layout",
     class = if (right) "sidebar-right",
     class = if (identical(sidebar$open, "closed")) "sidebar-collapsed",
-    `data-bslib-sidebar-init` = sidebar_id,
+    `data-bslib-sidebar-init` = sidebar_init,
     `data-bslib-sidebar-open` = sidebar$open,
-    `data-bslib-sidebar-border` = if (!is.null(border)) tolower(border),
-    `data-bslib-sidebar-border-radius` = if (!is.null(border_radius)) tolower(border_radius),
+    `data-bslib-sidebar-border` = sidebar_border,
+    `data-bslib-sidebar-border-radius` = sidebar_border_radius,
```

Comment:
For the sake of keeping the diff on R code minimal (especially since this has now been ported to Python), can we keep it the way it was?

Reply:
Done in 2f981db12393e52ff8a0fb5b4f1e45512ca3417b